### ISSUE TITLE
[❓Question ]  How to prevent model.predict() for unseen wrong values? 

### ISSUE THREAD

**Issue by VinayChaudhari1996** [pycaret/pycaret#1060]
#### Pycaret version :  2.2.3
____
Hello Guys ,

I have trained model on diamond dataset (Regression).

Now I'm trying to predict some records using trained model. I observe model is really good for unseen new data.

#### ⚠️ But when I give wrong input values to model it returns some prediction. But those values are not consider while training. 

 So how can I prevent this issue for unknown input values to (Regression/Classification) models. 

➡️Expected Result :- 
If we pass wrong data (values)  , model.predict should return some error or something.

Eg :- Error : Input data[some column name]  didn't observe while training. Unknown input found.

Help me to solve this issue.

Thanks ,

**Comment by Yard1** [comment]
Can you post the code you are using?

**Comment by pycaret** [comment]
@VinayChaudhari1996 Let's say you train your model on columns A, B, C, D, and then when using `predict_model`, you passed columns A, B, C, D, E, F. PyCaret will ignore column E and F internally. This means the predicted number is only based on A, B, C, D columns. 

On the contrary, if you train your model on columns A, B, C, D and pass columns A, B, D only when using the `predict_model`, It will return you an error saying that column is missing.

Hope this helps.

**Comment by VinayChaudhari1996** [comment]
> Can you post the code you are using?
Pycaret : v2.2.3
Model : dt
Dataset : Regression
____

![image](https://user-images.githubusercontent.com/42869040/107652236-5e748e80-6ca6-11eb-8dec-0b3740909cc5.png)

1) See I have added some random input values (XXX), but still, I get the prediction, How Can I prevent that any error or something "Unknown" found at ...

Thanks ,

**Comment by Yard1** [comment]
The columns you have added are simply ignored by the model.

**Comment by VinayChaudhari1996** [comment]
> The columns you have added are simply ignored by the model.

Yes , I know 
I'm talking about values 
( Check those random values for given key XXXX)

**Comment by Yard1** [comment]
They are not used in prediction - the label and score column are simply generated and appended to the dataframe you passed, but the columns that were not used during training are ignored.

**Comment by VinayChaudhari1996** [comment]
> They are not used in prediction - the label and score column are simply generated and appended to the dataframe you passed, but the columns that were not used during training are ignored. We have decided to do that instead of throwing an exception.

I consider this 2 columns for training.
=> Report and Polish

What I'm trying to say , 
Let's say Report has only 3 categorical values (A,B,C) 

And Polish :- ( X,Y,Z)

Then while prediction I'm trying to give different categorical values to that column. 

Report : XXXX <- Which is not appear in whole column while training.
(Same for polish)

So my question is how to catch this issue ?

**Comment by Yard1** [comment]
Is it possible for you to create a notebook that would show us the entire process, from creating the model to prediction?

**Comment by VinayChaudhari1996** [comment]
Yeah Sure,

Here you go

[pycaret_2_2_3.zip](https://github.com/pycaret/pycaret/files/5967568/pycaret_2_2_3.zip)

Thanks,

**Comment by Yard1** [comment]
Ok, I see what you mean now. By default, PyCaret will treat unseen levels in data similarly to missing values and impute them. This is controlled by arguments in `setup()`:
```
    handle_unknown_categorical: bool = True,
    unknown_categorical_method: str = "least_frequent",
```
However, if `handle_unknown_categorical` is set to `False`, there will be no exception as there should be, so I am marking this as a bug. Thanks for brining this to our attention.

**Comment by pycaret** [comment]
@Yard1 Are you gonna raise an exception when a new level is seen in categorical features when `handle_unknown_category` is set to False?

**Comment by Yard1** [comment]
I think that would be the best way to handle it @pycaret. Otherwise it will give stupid predictions. I believe that's how sklearn handles it too.

**Comment by Yard1** [comment]
Fixed in commit caa2eaf06a0896232d2ddf223b72103bba1e3a0f
